### DISCUSSION TITLE
List of possible improvements

### DISCUSSION THREAD

**Discussion by henrikwirth** [wp-graphql/wp-graphql#1258] [Ideas]
As I'm creating a theme based on WordPress Twenty Twenty I already ran into some things, that could be improved. I wanna use this issue, to gather all the trouble I run into and all the ideas that come up, to ease the development of themes.

I will update it step-by-step and let you know, when I finished.

1. Under `Reading Settings > Your homepage displays` It would be nice to get the `readingSettings` for what is set as `postsPage` and `frontPage` like so.
```graphql
readingSettings {
  frontPage
  postsPage
}
```
If `Your latest posts` is selected. Both could be null or `"/"`. There could be even more useful options like so:

```graphql
readingSettings {
  frontPage
  postsPage
  showFullText  # Weather the posts should be shown in full, or only an excerpt 
}
```

2. Some of the Author names are always null:
```graphql
{
  "data": {
    "allWpPost": {
      "nodes": [
        {
          "title": "Page 1",
          "author": {
            "name": "Henrik Wirth",
            "nickname": null,
            "nicename": null,
            "username": null,
            "firstName": "Henrik",
            "lastName": "Wirth",
            "uri": "author/admin/"
          },  
      ]
    }
  }
}
```

3. @TylerBarnes It would be nice to have an Option in the WordPress plugin to deactivate the `Preview` override. ** has been implemented already **
4. dateGmt and date seem to have the same output:
```
{
   "title": "Page 1",
  "date": "2020-04-22T14:02:06",
  "dateGmt": "2020-04-22T14:02:06"
},
```
5. Date Formatting would be a nice thing. WP comes with Date Format options. So either a field coming from WPGQL that is formatted according to WP or some resolver on the Gatsby-source-plugin side would be nice so this doesn't have to happen in the JS frontend and rather on build time.
6. Encoding of special characters in titles and other fields is troublesome. For that it would be nice to define specific fields, which get encoded on build-time or WP-API so it again isn't required on the JS-frontend.
7. I'm sure this is already planned, but it would be nice to have the status of Gatsby Cloud or Netlify somewhere in the WP Admin Bar to see how things work out with the incremental build.
8. When permalink structure is changed, it doesn't trigger Gatsby Cloud. Also when I triggered rebuild with clearing cache on Gatsby Cloud I still had the pages on the old permalinks remaining. (Maybe server cache?)
9. WpAvatar is not processed in gatsby-image
10. It would be nice to have an easy switch to turn on/off the direct incremental builds on Posts/Page updates. (UseCase: If you only have 25 Live updates a day on a free plan, you might wanna have control when they are used.)
11. Gatsby Cloud rebuilds 2x (@TylerBarnes explained this is due to Gutenberg sending 2 post updates) Maybe it would be an option to collect post updates and put them in an update stack? If there is 2 of the same only one will get send. Delaying when the update stack is processed would help to have enough time to compare the queue items. (just an idea ...) could be on the Gatsby Cloud side too.
12. Author's description field is missing in gatsby-wordpress-source. **Figured** this is part of the default setting. Now it works. But I noticed, that Gatsby is not updating the description when it is change (on `gatsby develop`)
13. Although I'm not a big fan of widgets. It is something that is missing and probably should be supported in a way people can extend the widget schema easily

**Comment by CodeProKid** [comment]
@henrikwirth once you get a more finalized list of items, I think it would be good to break them out into individual tickets so we can triage them individually.

**Comment by henrikwirth** [comment]
@CodeProKid ya definitely. That was the plan. Also some stuff is already fixed in the meantime.

**Comment by jasonbahl** [comment]
@henrikwirth I'm labeling this as close candidate, as actionable items should be moved to their own issues for tracking. 

I love all the ideas and sharing them here to start! Good way to get some conversations going and determine what _needs_ a separate issue, what's already in progress, what should be a plugin, etc.

**Comment by jasonbahl** [comment]
@henrikwirth now that we have Github Discussions, I'm going to move this over there. I'd like to keep discussing these ideas, but would prefer actionable issues once ready.
